lib/src/util/event_handler_error_catcher.dart

```diff
@@ -0,0 +1,45 @@
+// @dart = 2.7
+
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:html';
+
+import 'package:react/react_client/react_interop.dart' show throwErrorFromJS;
+
+/// INTENDED FOR INTERNAL USE IN REACT_TESTING_LIBRARY ONLY.
+///
+/// Event Handlers triggered by a UserEvent or fireEvent may throw errors
+/// but due to how the DOM natively works with Events, the point that dispatched
+/// the event will never surface it, leading to false positives.
+///
+/// This function wraps around the call to the JS RTL within UserEvent and
+/// fireEvent in order to surface the errors.
+T eventHandlerErrorCatcher<T>(T Function() cb) {
+  T result;
+  final errors = <dynamic>{};
+  final errorStream = window.onError.cast<ErrorEvent>().listen((event) {
+    errors.add(event.error);
+  });
+  result = cb();
```

Comment:
I think if `cb` returns a future, we should wait until the future is completed before cancelling the subscription.

For example, if `keyboardWithDelay` throws an exception on the second keystroke and not the first. (Could we add a test for that?)

Reply:
what sucks about this is, it does work. but my test suite just like tests one error then multiple, nothing except typing had multiple possible right out the gate lol even in the typeWithDelay test i only typed one letter to not break the first test lol, ill add a typeWithDelay test that shows it working.

Reply:
It does? I would expect it to not work since the stream subscription would get torn down synchronously while the keyboard events are still firing asychronously